### PR TITLE
SOLVED Issue #3436: Default file extension should be .py 

### PR DESCRIPTION
**Pull request by malliwi88**
Now it's possible select file extension when Save As files in Spyder Editor @goanpeca

### CODE REVIEW

File: spyder/plugins/editor.py [hunk 1]
```diff
@@ -1887,14 +1887,47 @@ def save(self, index=None, force=False):
...
-            fname = editorstack.get_current_filename()
-            if CONF.get('workingdir', 'editor/save/auto_set_to_basedir'):
-                self.open_dir.emit(osp.dirname(fname))
-            self.__add_recent_file(fname)
+        #results = editorstack.get_todo_results()
+
+        if index is None:
+            index = editorstack.get_stack_index()
+
+        finfo = editorstack.data[index]
```
Comment (goanpeca) [inline spyder/plugins/editor.py]: Why is all this code here? There was a save_as method already on the editor widget, located at 
https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/editor.py#L1310

Why not make the changes there?

File: spyder/config/utils.py [hunk 2]
```diff
@@ -119,7 +119,7 @@ def get_edit_filetypes():
     other_exts = ['.ipynb', '.md']
     all_exts = tuple(pygments_exts + other_exts)
     text_filetypes = (_("Supported text files"), all_exts)
-    return [text_filetypes] + EDIT_FILETYPES
+    return EDIT_FILETYPES + [text_filetypes]
```
Comment (malliwi88) [inline spyder/config/utils.py]: Hi Gonzalo, I decided to switch line (#122) so Spyder will suggest in comboBox to save the file with the extensions .py, .pyw, .ipv instead *.6pl, *.6pm, *:ABAP

I consider the most programmers that use Spyder need this suggestion by default instead search it at last of ComboBox  extension list.

**Comment by goanpeca** [comment]
@malliwi88 you are not supposed to merge prs without a final review. The merge action is at this moment only executed by Carlos, Nodd, Jitae or myself. Please never do this again!

**Comment by malliwi88** [comment]
Okay, sorry for the mistake @goanpeca

File: spyder/app/mainwindow.py [hunk 3]
```diff
@@ -1231,9 +1231,9 @@ def post_visible_setup(self):
 
         # Hide Internal Console so that people don't use it instead of
         # the External or IPython ones
-        if self.console.dockwidget.isVisible() and DEV is None:
-            self.console.toggle_view_action.setChecked(False)
-            self.console.dockwidget.hide()
+        #if self.console.dockwidget.isVisible() and DEV is None:
```
Comment (goanpeca) [inline spyder/app/mainwindow.py]: @malliwi88, please restore this behavior, this should not have changed.

File: spyder/widgets/editor.py [hunk 4]
```diff
@@ -1309,12 +1311,22 @@ def select_savename(self, original_filename):
...
         finfo = self.data[index]
-        filename = self.select_savename(finfo.filename)
+
+        self.redirect_stdio.emit(False)
+        filename, _selfilter = getsavefilename(self, _("Save file"),
+                                               finfo.filename, get_edit_filters())
+        self.redirect_stdio.emit(True)
+
         if filename:
+            print("%s %s" % (finfo, index))
```
Comment (goanpeca) [inline spyder/widgets/editor.py]: Please remove this

File: spyder/widgets/editor.py [hunk 5]
```diff
@@ -1325,10 +1337,6 @@ def save_as(self, index=None):
 
             new_index = self.rename_in_data(index, new_filename=filename)
 
-            # We pass self object ID as a QString, because otherwise it would
```
Comment (goanpeca) [inline spyder/widgets/editor.py]: Why did you remove these comments?

File: spyder/widgets/editor.py [hunk 6]
```diff
@@ -1309,12 +1311,22 @@ def select_savename(self, original_filename):
...
+                return
             index = self.get_stack_index()
+
         finfo = self.data[index]
-        filename = self.select_savename(finfo.filename)
+
+        self.redirect_stdio.emit(False)
+        filename, _selfilter = getsavefilename(self, _("Save file"),
+                                               finfo.filename, get_edit_filters())
+        self.redirect_stdio.emit(True)
```
Comment (goanpeca) [inline spyder/widgets/editor.py]: We need a smarter way so that if I click on save us, by default I should get the selection to be the actual file extension I have.

You can make a test of this by creating a .txt or .yaml file and then click on save as....  we should not get 

somefile.txt.py or somefile.yaml... we should get somefile.txt and the appropriate selection on the combobox

**Comment by ccordoba12** [comment]
@goanpeca, why did you merge this one without @malliwi88 addressing your review?

Besides, this one went to master instead of 3.x, so now we need to backport it ;-p

**Comment by ccordoba12** [comment]
Sorry, it seems it was @malliwi88 who merged this one. @malliwi88, is that right?

**Comment by goanpeca** [comment]
Yes @ccordoba12, @malliwi88 accidentally merged it.

**Comment by ccordoba12** [comment]
Ups! Ok :-)

Should we revert it?

**Comment by goanpeca** [comment]
Dont know what is better.... can we revert it so we can start from scratch and on the right branch?

**Comment by ccordoba12** [comment]
Yeah, please revert it use the button above.

@malliwi88, don't worry about this, but please try to be more careful for the next one :-)
